azure-pipelines.yml

```diff
@@ -0,0 +1,103 @@
+strategy:
+  matrix:
+    windows-msvc-2019:
+      IMAGE_NAME: windows-2019
+      BUILD_CONFIG: Release
+      RUN_PCM_TESTS: ON
+      CMAKE_ENV:
+      INIT_CMD:
+
+    windows-msvc-2017:
+      IMAGE_NAME: vs2017-win2016
+      BUILD_CONFIG: Release
+      RUN_PCM_TESTS: ON
+      CMAKE_ENV:
+      INIT_CMD:
+
+    macOS-xcode-12.2:
+      IMAGE_NAME: macOS-10.15
+      BUILD_CONFIG: Release
+      RUN_PCM_TESTS: ON
+      CMAKE_ENV:
+      INIT_CMD: xcversion select 12.2
+
+    macOS-xcode-11.7:
+      IMAGE_NAME: macOS-10.15
+      BUILD_CONFIG: Release
+      RUN_PCM_TESTS: ON
+      CMAKE_ENV:
+      INIT_CMD: xcversion select 11.7
+
+    ubuntu-gcc-7:
+      IMAGE_NAME: ubuntu-18.04
+      BUILD_CONFIG: Release
+      RUN_PCM_TESTS: ON
+      CMAKE_ENV: CC=gcc-7 CXX=g++-7
+      INIT_CMD:
+
+    ubuntu-gcc-8:
+      IMAGE_NAME: ubuntu-18.04
+      BUILD_CONFIG: Release
+      RUN_PCM_TESTS: ON
+      CMAKE_ENV: CC=gcc-8 CXX=g++-8
+      INIT_CMD:
+
+    ubuntu-gcc-9:
+      IMAGE_NAME: ubuntu-18.04
+      BUILD_CONFIG: Release
+      RUN_PCM_TESTS: ON
+      CMAKE_ENV: CC=gcc-9 CXX=g++-9
+      INIT_CMD:
+
+    ubuntu-gcc-9-gcov:
+      IMAGE_NAME: ubuntu-18.04
```

Comment:
Should we add a strategy for 20.04 LTS, too?

Reply:
20.04 is still in preview state see https://docs.microsoft.com/en-us/azure/devops/pipelines/agents/hosted?view=azure-devops&tabs=yaml. 18.04 is the latest official image so i'd rather stick with that for now